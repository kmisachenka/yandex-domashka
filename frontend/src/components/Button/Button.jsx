import React from 'react';
import * as PropTypes from 'prop-types';
import cn from 'classnames';
import styles from './Button.module.scss';

export default function Button(props) {
  const { children, theme, disabled, handleClick } = props;
  const themeStyles = styles[theme];
  return (
    <button
      className={cn(styles.button, themeStyles)}
      disabled={disabled}
      type="button"
      onClick={handleClick}
    >
      {children}
    </button>
  );
}

Button.defaultProps = {
  theme: 'default',
  disabled: false,
  handleClick: () => {},
};

Button.propTypes = {
  children: PropTypes.string.isRequired,
  theme: PropTypes.oneOf(['default', 'primary']),
  disabled: PropTypes.bool,
  handleClick: PropTypes.func,
};
