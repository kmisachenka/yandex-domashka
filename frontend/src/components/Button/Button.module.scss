.button {
  appearance: none;
  border: 0;

  border-radius: 4px;
  color: #333;

  font-family: Helvetica, serif;
  font-size: 16px;
  line-height: 32px;
  margin: 0;
  min-height: 32px;
  padding: 0 15px;
  text-align: center;

  width: 100%;

  &:hover {
    &:not(:disabled) {
      background-color: #3ba6ff;
      color: #fff;
      transition: background-color 0.2s ease, color 0.1s ease-in;
    }

    &:active {
      background-color: #008000;
      color: #fff;
    }
  }

  &:disabled {
    color: rgba(51, 51, 51, 0.6);
  }
}

.primary {
  background: #fc0;
}
