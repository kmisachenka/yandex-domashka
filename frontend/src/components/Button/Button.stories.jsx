import React from 'react';
import { storiesOf } from '@storybook/react';

import Button from './Button';

storiesOf('Кнопка', module)
  .add('с текстом', () => (
    <Button>Активные</Button>
  ))
  .add('неактивная', () => (
    <Button disabled>Активные</Button>
  ))
  .add('с темой primary', () => (
    <Button theme="primary">Добавить</Button>
  ));
