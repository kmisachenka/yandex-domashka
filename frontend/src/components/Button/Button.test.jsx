import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from './Button';

describe('Button', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Button>button</Button>);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render children', () => {
    const wrapper = shallow(<Button>Test Button</Button>);
    expect(wrapper).toHaveText('Test Button');
  });
  it('should render default theme by default', () => {
    const wrapper = shallow(<Button>button</Button>);
    expect(wrapper).toHaveClassName('default');
  });
  it('should render primary theme', () => {
    const wrapper = shallow(<Button theme="primary">button</Button>);
    expect(wrapper).toHaveClassName('primary');
  });
  it('should handle click', () => {
    const handleClick = jest.fn();
    const wrapper = mount(<Button handleClick={handleClick}>Test</Button>);
    const button = wrapper.find('.button');
    button.simulate('click');
    expect(handleClick).toHaveBeenCalled();
  });
});
