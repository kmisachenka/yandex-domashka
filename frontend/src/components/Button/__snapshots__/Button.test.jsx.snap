// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
<button
  className="button default"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  button
</button>
`;
